magpie/ui/management/views.py

```diff
@@ -329,14 +330,17 @@ def edit_user(self):
             if "clean_resource" in self.request.POST:
                 # "clean_resource" must be above "edit_permissions" because they"re in the same form.
                 self.delete_resource(res_id)
-            elif "edit_permissions" in self.request.POST:
-                if not res_id or res_id == "None":
-                    remote_id = int(self.request.POST.get("remote_id"))
-                    services_names = [s["service_name"] for s in services.values()]
-                    res_id = self.add_remote_resource(cur_svc_type, services_names, user_name, remote_id, is_user=True)
-
-                removed_perms, new_perms = \
-                    self.edit_user_or_group_resource_permissions(user_name, res_id, is_user=True)
+            elif "edit_permissions" in self.request.POST and not inherit_grp_perms:
+                # FIXME:
```

Comment:
Why doing stuff in batch remove the need to add remote ressources?
If I want to add a permission on something that doesn't already exist, it seems like the add_remote fct should be required, no?

Reply:
This is because david c. did that part, but the trick is that the ui shows ressources that doesn't exist in magpie db. It is using the synched remote ressources db (if I remember correctly) and mix it with the magpie ressources. 
So the synch doesn't really populate the magpie ressources, but the remote db and the mix occurs in real time for the ui. So checking a ressource doesn't mean that it already exists. If that ressource came from the remote db it must first be added to the magpie db before setting its permission.

Reply:
We will need to revisit this whole functionality IMHO. It is not very well maintained. 
For example, we cannot update the service type at later date using the API, and creating a service via the UI doesn't provide any selector to specify which one we want.
I have also booted a quick demo THREDDS server and it fails to sync files (with force sync button) due to internal parsing.